### ISSUE TITLE
I can't login with sign up password.

### ISSUE THREAD

**Issue by mzch** [Erudika/scoold#283]
I signed up scoold and activate my account normally, but I can't login.

Scoold: `1.47.1`
Para: `1.43.2`

- Scoold log
```
2021-12-30 08:36:53,946 [INFO ] com.erudika.scoold.ScooldServer - Starting ScooldServer v1.47.1 using Java 11.0.13 on n16 with PID 52538 (/srv/scoold/scoold-1.47.1.jar started by scoold in /srv/scoold)
2021-12-30 08:36:53,973 [INFO ] com.erudika.scoold.ScooldServer - The following profiles are active: production
2021-12-30 08:36:57,129 [INFO ] com.erudika.scoold.ScooldServer - Listening on port 8000...
2021-12-30 08:36:57,809 [INFO ] com.erudika.scoold.ScooldServer - Initialized ParaClient with endpoint https://paraenv.com and access key 'app:scoold'.
2021-12-30 08:37:02,256 [INFO ] com.erudika.scoold.ScooldServer - Started ScooldServer in 9.427 seconds (JVM running for 11.435)
2021-12-30 08:38:29,336 [ERROR] com.erudika.para.client.ParaClient - 400 - 400 - Failed to authenticate user with 'password'. Check if user is active.
2021-12-30 08:38:57,009 [ERROR] com.erudika.para.client.ParaClient - 400 - 400 - Failed to authenticate user with 'password'. Check if user is active.
2021-12-30 08:40:08,438 [ERROR] com.erudika.para.client.ParaClient - 400 - 400 - Failed to authenticate user with 'password'. Check if user is active.
```

- Para log
```
2021-12-30 08:36:34,426 [INFO ] com.erudika.para.core.utils.Para - --- Para.initialize() [production] ---
2021-12-30 08:36:34,436 [INFO ] c.erudika.para.core.utils.CoreUtils - Loaded new DAO, Search and Cache implementations - MongoDBDAO, LuceneSearch and CaffeineCache.
2021-12-30 08:36:34,706 [INFO ] c.e.p.s.persistence.MongoDBUtils - MongoDB host: 192.168.207.10:27017, database: para
2021-12-30 08:36:34,814 [INFO ] org.mongodb.driver.cluster - Cluster created with settings {hosts=[192.168.207.10:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
2021-12-30 08:36:35,074 [INFO ] org.mongodb.driver.cluster - Cluster description not yet available. Waiting for 30000 ms before timing out
2021-12-30 08:36:35,144 [INFO ] org.mongodb.driver.connection - Opened connection [connectionId{localValue:1, serverValue:4095}] to 192.168.207.10:27017
2021-12-30 08:36:35,144 [INFO ] org.mongodb.driver.connection - Opened connection [connectionId{localValue:2, serverValue:4096}] to 192.168.207.10:27017
2021-12-30 08:36:35,149 [INFO ] org.mongodb.driver.cluster - Monitor thread successfully connected to server with description ServerDescription{address=192.168.207.10:27017, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=14, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=90966815, setName='rs01', canonicalAddress=192.168.207.10:27017, hosts=[192.168.207.10:27017], passives=[], arbiters=[], primary='192.168.207.10:27017', tagSet=TagSet{[]}, electionId=7fffffff0000000000000014, setVersion=2, topologyVersion=TopologyVersion{processId=61be0f8fbda1702232c03e32, counter=6}, lastWriteDate=Thu Dec 30 08:36:34 PST 2021, lastUpdateTimeNanos=1035870096675766}
2021-12-30 08:36:35,470 [INFO ] org.mongodb.driver.connection - Opened connection [connectionId{localValue:3, serverValue:4097}] to 192.168.207.10:27017
2021-12-30 08:36:38,046 [INFO ] c.e.para.server.utils.HealthUtils - Server is healthy.
2021-12-30 08:36:38,189 [INFO ] c.e.p.server.metrics.MetricsUtils - Found root app 'para' and 1 existing child app(s).
2021-12-30 08:36:39,235 [INFO ] com.erudika.para.server.ParaServer - Starting ParaServer using Java 11.0.13 on n16 with PID 52483 (/srv/para/para-jar-1.43.2.jar started by para in /srv/para)
2021-12-30 08:36:39,237 [INFO ] com.erudika.para.server.ParaServer - The following profiles are active: production
2021-12-30 08:36:41,617 [INFO ] com.erudika.para.server.ParaServer - Instance #1 initialized and listening on http://localhost:8080
2021-12-30 08:36:45,298 [INFO ] com.erudika.para.server.ParaServer - Started ParaServer in 7.1 seconds (JVM running for 14.018)
2021-12-30 08:38:56,998 [WARN ] c.e.p.server.security.SecurityUtils - Account 61cde09fe5d25406607be93f (scoold/mzch@mzch.org) is locked.
2021-12-30 08:40:08,431 [WARN ] c.e.p.server.security.SecurityUtils - Account 61cde09fe5d25406607be93f (scoold/mzch@mzch.org) is locked.
```

- Database records (Hide personal info)
```
{
    _id: '61cde09fe5d25406607be93f',
    _ObjectId: '61cde09fe5d25406607be950',
    timestamp: 1640882335962,
    type: 'user',
    appid: 'scoold',
    name: 'ParaObject 1640882335831',
    votes: 0,
    version: 0,
    stored: true,
    indexed: true,
    cached: true,
    identifier: 'mzch@example.com',
    groups: 'users',
    active: false,
    twoFA: false,
    email: 'mzch@example.com',
    picture: 'https://www.gravatar.com/avatar/xxxx?size=400&d=mm&r=pg',
    tokenSecret: 'secret'
}
```

I clicked the activation link in my activation mail from my scoold. But `active` is still false. Is this a bug?

**Comment by albogdano** [comment]
I can't reproduce this issue. I'm using MongoDB 5.0.5 and latest Para + Scoold. Perhaps the activation link is incorrect. I am able to sign up and activate my user account.

**Comment by mzch** [comment]
I use MongoDB `5.1.0`.

**Comment by albogdano** [comment]
The latest Mongo is 5.0.5. There is no 5.1 yet.

**Comment by mzch** [comment]
See https://github.com/mongodb/mongo/releases/tag/r5.1.1

**Comment by albogdano** [comment]
If there are no breaking changes between Mongo 5.0.5 and 5.1, everything should work as expected. I am sure you have a problem with the email verification link and not Para or the Mongo plugin for Para.

**Comment by mzch** [comment]
Yes, I also think Para has no problem. But Scoold recorded no activation log. I think Scoold has a problem.

**Comment by albogdano** [comment]
Can you delete the `user` and `profile` objects for that user from MongoDB Compass and then try to sign up again? Also pay attention to the activation link - it should have the correct URL. Or click "Resend activation link" in Scoold.

**Comment by mzch** [comment]
1. Deleted alll records.
2. I signed up again.
3. Created `identifier` record, but `ParaObject 1640967052848` record was not created. `identifier` record is as below:
```
{
    _id: 'mzch@mzch.org',
    _ObjectId: '61cf2b8c96caaf5426a87255',
    timestamp: 1640967052848,
    type: 'sysprop',
    appid: 'scoold',
    creatorid: '61cf290f96caaf5426a8722c',
    name: 'identifier',
    votes: 0,
    version: 0,
    stored: true,
    indexed: true,
    cached: true,
    properties: {
        etoken: 'cVVWV1hiT2Q5dVJmOU9TS0NWc1BKdGZUajI1VENiVDZTZWhWazdTZVFLOD0',
        password: 'hashed-password'
    },
    updated: 1640967053699
}
```
4. Click the activation link. e.g. https://scoold.org/signin/register?id=61cf290f96caaf5426a8722c&token=cVVWV1hiT2Q5dVJmOU9TS0NWc1BKdGZUajI1VENiVDZTZWhWazdTZVFLOD0
5. I tried to sign in, but failed. As the result, `ParaObject 1640967052848` record was created. At this time etoken object was deleted from `identifier` record.
```
{
    _id: 'mzch@mzch.org',
    _ObjectId: '61cf2bfd96caaf5426a87273',
    timestamp: 1640967165518,
    type: 'sysprop',
    appid: 'scoold',
    creatorid: '61cf2bfc96caaf5426a87256',
    name: 'identifier',
    votes: 0,
    version: 0,
    stored: true,
    indexed: true,
    cached: true,
    properties: {
        password: 'hashed-password'
    }
}
```
6. I tried to resend confirmation mail, but no mail has been sent.

**Comment by albogdano** [comment]
What do you see in the browser after you click the activation link? Anything in the logs? Everything works well on my machine. Maybe you need to update your MongoDB plugin for Para - latest version is 1.37.1.

**Comment by mzch** [comment]
When I clicked an activation link, the sign in screen was shown. Email verified screen isn't shown.
And I tried to use PostgreSQL as backend, so I was able to sign up and login successfully.
Perhaps only when using MongoDB, sign up process fails...

**Comment by albogdano** [comment]
You are definitely doing something wrong, because I double checked everything and it all works. 

- check that the `_id` of the `Sysprop` identifier object and you email address are the same.
- check that the `token` parameter in the activation link is the same as it is in the identifier object at `properties.etoken`.
- click the resend activation link to test again
- configure SMTP to actually send you the emails

I am closing this for now - I see nothing wrong here.

**Comment by albogdano** [comment]
I found the issue and fixed it in `1.47.2`. Indeed, there was a problem with the email verification link.
